package/MDAnalysis/analysis/pca.py

```diff
@@ -414,6 +414,130 @@ def transform(self, atomgroup, n_components=None, start=None, stop=None,
+
+        Examples
+        --------
+        Run PCA class before using this function. For backbone PCA, run::
+
+            pca = PCA(universe, select='backbone').run()
+
+        Obtain a transformation function to project the
+        backbone trajectory onto the first principal component::
+
+            project = pca.project_single_frame(components=0)
+
+        To projet onto the first two components, run::
+
+            project = pca.project_single_frame(components=[0,1])
+
+        Alternatively, the transformation can be applied to PCA atoms and
+        extrapolated to other atoms according to the CA atom's translation
+        in each residue::
+
+            all = u.select_atoms('all')
+            project = pca.project_single_frame(components=0,
+                                               group=all, anchor='name CA')
+
+        Finally, apply the transformation function to a timestep::
+
+            project(u.trajectory.ts)
+
+        or apply the projection to the universe::
+
+            u.trajectory.add_transformations(project)
+
+
+        .. versionadded:: 2.2.0
+        """
+        if not self._calculated:
+            raise ValueError('Call run() on the PCA before projecting')
+
+        if group is not None:
+            if anchor is None:
+                raise ValueError("'anchor' cannot be 'None'" +
+                                 " if 'group' is not 'None'")
+
+            anchors = group.select_atoms(anchor)
+            anchors_res_ids = anchors.resindices
+            if np.unique(anchors_res_ids).size != anchors_res_ids.size:
+                raise ValueError("More than one 'anchor' found in residues")
+            if not np.isin(group.resindices, anchors_res_ids).all():
+                raise ValueError("Some residues in 'group'" +
+                                 " do not have an 'anchor'")
+            if not anchors.issubset(self._atoms):
+                raise ValueError("Some 'anchors' are not part of PCA class")
+
+            non_pca = group - self._atoms
+            pca_res_indices, pca_res_counts = np.unique(
+                self._atoms.resindices, return_counts=True)
+
+            # matrix_extrapolate is later multiplied to anchors' coordinates.
+            # multiplication shapes the array as appropriate for non_pca atoms
+            matrix_extrapolate = np.zeros((non_pca.n_atoms, anchors.n_atoms))
+            cumul_atoms_old = 0
+            for res in group.residues:
+                n_common = pca_res_counts[np.where(
+                           pca_res_indices == res.resindex)][0]
+                cumul_atoms_new = (cumul_atoms_old +
+                                   res.atoms.n_atoms - n_common)
+                matrix_extrapolate[cumul_atoms_old:cumul_atoms_new,
+                                   res.resindex] = 1
+                cumul_atoms_old = cumul_atoms_new
+
+        if components is None:
+            components = range(self.results.p_components.shape[1])
```

Comment:
Using a `range` object works for indexing a np.array but is ~50x slower for me. I'd do
```suggestion
            components = np.arange(self.results.p_components.shape[1])
```
and store the array.

```python
In [1]: import numpy as np

In [2]: a = np.arange(100)

In [3]: a[range(4)]
Out[3]: array([0, 1, 2, 3])

In [4]: a[range(29)]
Out[4]:
array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28])

In [5]: %timeit a[range(29)]
9.29 µs ± 61.7 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [6]: ix = np.arange(29)

In [7]: %timeit a[ix]
209 ns ± 7.5 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [8]: 9.29e-6/209e-9
Out[8]: 44.449760765550245
```

Reply:
Fixed. I think, it was a typo! An oversight that _seemed_ to work.